bin/network-monitor/src/deploy/mod.rs

```diff
@@ -0,0 +1,333 @@
+
+    assembler
+        .clone()
+        .assemble_library([module])
+        .map_err(|e| anyhow::anyhow!("Failed to assemble library: {e}"))
+}
+
+// MONITOR DATA STORE
+// ================================================================================================
+
+/// A [`DataStore`] implementation for the network monitor.
+pub struct MonitorDataStore {
+    wallet_account: Mutex<Account>,
+    counter_account: Mutex<Account>,
+    #[allow(dead_code)]
+    wallet_init_seed: Option<Word>,
+    block_header: BlockHeader,
+    partial_block_chain: PartialBlockchain,
+    mast_store: TransactionMastStore,
+}
+
+impl MonitorDataStore {
+    pub fn new(
+        wallet_account: Account,
+        counter_account: Account,
+        wallet_init_seed: Option<Word>,
+        block_header: BlockHeader,
+        partial_block_chain: PartialBlockchain,
+    ) -> Self {
+        let mast_store = TransactionMastStore::new();
+        mast_store.insert(wallet_account.code().mast());
+        mast_store.insert(counter_account.code().mast());
+
+        Self {
+            mast_store,
+            wallet_account: Mutex::new(wallet_account),
+            counter_account: Mutex::new(counter_account),
+            wallet_init_seed,
+            block_header,
+            partial_block_chain,
+        }
+    }
+}
+
+impl DataStore for MonitorDataStore {
+    async fn get_transaction_inputs(
+        &self,
+        account_id: AccountId,
+        mut _block_refs: BTreeSet<BlockNumber>,
+    ) -> Result<(PartialAccount, BlockHeader, PartialBlockchain), DataStoreError> {
+        let account = self.wallet_account.lock().await;
+
+        let account = if account_id == account.id() {
+            account.to_owned()
+        } else {
+            self.counter_account.lock().await.to_owned()
+        };
+
+        Ok(((&account).into(), self.block_header.clone(), self.partial_block_chain.clone()))
+    }
+
+    async fn get_storage_map_witness(
+        &self,
+        account_id: AccountId,
+        map_root: Word,
+        map_key: Word,
+    ) -> Result<miden_objects::account::StorageMapWitness, DataStoreError> {
+        let account = self.wallet_account.lock().await;
+
+        let account = if account_id == account.id() {
+            account.to_owned()
+        } else {
+            self.counter_account.lock().await.to_owned()
+        };
+
+        let mut map_witness = None;
+        for slot in account.storage().slots() {
+            if let StorageSlot::Map(map) = slot {
+                if map.root() == map_root {
+                    map_witness = Some(map.open(&map_key));
+                }
+            }
+        }
+
+        if let Some(map_witness) = map_witness {
+            Ok(map_witness)
+        } else {
+            Err(DataStoreError::Other {
+                error_msg: "account storage does not contain the expected root".into(),
+                source: None,
+            })
+        }
```

Comment:
In `get_transaction_inputs` you are instantiating the partial accounts with `PartialAccount::from`. If you include all the account data (I think you'd need to use `PartialAccount::new`), I think you should be able to remove the implementations of these methods. I would test removing them. We can also keep them, but it's more maintenance overhead + more code to test.